streamalert/shared/firehose.py

```diff
@@ -334,6 +334,33 @@ def generate_firehose_name(cls, prefix, log_stream_name):
             base_name, hashlib.md5(stream_name.encode()).hexdigest()  # nosec
         )[:cls.AWS_FIREHOSE_NAME_MAX_LEN]
 
+    @classmethod
+    def artifacts_firehose_stream_name(cls, config):
+        """Return Artifacts Firehose Stream Name
+
+        Args:
+            config (dict): The loaded config from the 'conf/' directory
+
+        Returns:
+            str: Artifacts Firehose Stream Name
+        """
+        stream_name = config.get('lambda', {}).get(
+            'artifact_extractor_config', {}
+        ).get('firehose_stream_name')
+
+        if stream_name:
+            # support custom firehose stream name of Artifacts. User should make sure the length of
+            # the custom firehose name is no longer than 64 chars, otherwise the firehose will be
+            # failed to create. StreamAlert is not responsible for checking for custom firehose name
+            # since it should not change custom settings.
+            return stream_name
+
+        return cls.generate_firehose_name(
```

Comment:
remove the above `if stream_name:` check just do:

```suggestion
        return stream_name or cls.generate_firehose_name(
```

Reply:
Neat!